bulk_indexer.go

```diff
@@ -91,6 +91,13 @@ type BulkIndexerConfig struct {
 	//
 	// RequireDataStream is disabled by default.
 	RequireDataStream bool
+
+	// PopulateFailedDocsSource controls whether each BulkIndexerResponseItem.Source
+	// in BulkIndexerResponseStat.FailedDocs is populated with the source of the item,
+	// which includes the action line and the document line.
+	// For testing and debugging only. Use with caution as it may expose sensitive data.
+	// Any clients should relay this warning to users.
```

Comment:
```suggestion
	//
	// WARNING: this is provided for testing and debugging only.
	// Use with caution as it may expose sensitive data; any clients
	// of go-docappender enabling this should relay this warning to
	// their users. Setting this will also add memory overhead.
```

Couple of suggestions:
 - make the warning a bit more prominent
 - mention that this will also impact memory usage (further discouraging use, and avoiding surprises)